keys.go

```diff
@@ -97,52 +95,46 @@ func (keys ServerKeys) PublicKey(keyID KeyID, atTS Timestamp) []byte {
 
 // FetchKeysDirect fetches the matrix keys directly from the given address.
 // Optionally sets a SNI header if ``sni`` is not empty.
-// Returns the server keys and the state of the TLS connection used to retrieve them.
-func FetchKeysDirect(serverName ServerName, addr, sni string) (*ServerKeys, *tls.ConnectionState, error) {
-	// Create a TLS connection.
-	tcpconn, err := net.Dial("tcp", addr)
-	if err != nil {
-		return nil, nil, err
+// Optionally sets a timeout to the HTTP client if ``timeout`` isn't 0.
+// Note that this function doesn't check the validity of the certificate(s)
+// served by the server.
+// Returns the server keys and the state of the TLS connection used to retrieve
+// them.
+func FetchKeysDirect(addr, sni string, timeout time.Duration) (*ServerKeys, *tls.ConnectionState, error) {
```

Comment:
can you document the params?

Reply:
```golang
// FetchKeysDirect fetches the matrix keys directly from the given address.
// Optionally sets a SNI header if ``sni`` is not empty.
// Optionally sets a timeout to the HTTP client if ``timeout`` isn't 0.
// Note that this function doesn't check the validity of the certificate(s)
// served by the server.
// Returns the server keys and the state of the TLS connection used to retrieve
// them.
```

Isn't there already enough documentation, since it documents `addr`, `sni` and `timeout`?

Reply:
sorry yes, I guess there is. I didn't really read it properly.